geode-for-redis/src/integrationTest/java/org/apache/geode/redis/internal/commands/executor/list/AbstractLIndexIntegrationTest.java

```diff
@@ -0,0 +1,146 @@
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.geode.redis.internal.commands.executor.list;
+
+import static org.apache.geode.redis.RedisCommandArgumentsTestHelper.assertExactNumberOfArgs;
+import static org.apache.geode.redis.internal.RedisConstants.ERROR_NOT_INTEGER;
+import static org.apache.geode.redis.internal.RedisConstants.ERROR_WRONG_TYPE;
+import static org.apache.geode.test.dunit.rules.RedisClusterStartupRule.BIND_ADDRESS;
+import static org.apache.geode.test.dunit.rules.RedisClusterStartupRule.REDIS_CLIENT_TIMEOUT;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import java.util.concurrent.atomic.AtomicReference;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import redis.clients.jedis.HostAndPort;
+import redis.clients.jedis.JedisCluster;
+import redis.clients.jedis.Protocol;
+
+import org.apache.geode.redis.ConcurrentLoopingThreads;
+import org.apache.geode.redis.RedisIntegrationTest;
+
+public abstract class AbstractLIndexIntegrationTest implements RedisIntegrationTest {
+  private static final String NON_EXISTENT_LIST_KEY = "{tag1}nonExistentKey";
+  private static final String LIST_KEY = "{tag1}listKey";
+  private static final String[] LIST_ELEMENTS =
+      {"aardvark", "bats", "chameleon", "deer", "elephant", "flamingo", "goat"};
+  private JedisCluster jedis;
+
+  @Before
+  public void setUp() {
+    jedis = new JedisCluster(new HostAndPort(BIND_ADDRESS, getPort()), REDIS_CLIENT_TIMEOUT);
+  }
+
+  @After
+  public void tearDown() {
+    flushAll();
+    jedis.close();
+  }
+
+  @Test
+  public void lindex_wrongNumberOfArgs_returnsError() {
+    assertExactNumberOfArgs(jedis, Protocol.Command.LINDEX, 2);
+  }
+
+  @Test
+  public void lindex_withPositiveIndex_withNonExistentList_returnsNull() {
+    assertThat(jedis.lindex(NON_EXISTENT_LIST_KEY, 2)).isNull();
+  }
+
+  @Test
+  public void lindex_withNegativeIndex_withNonExistentList_returnsNull() {
+    assertThat(jedis.lindex(NON_EXISTENT_LIST_KEY, -2)).isNull();
+  }
+
+  @Test
+  public void lindex_withPositiveIndexes_returnsElement() {
+    jedis.lpush(LIST_KEY, LIST_ELEMENTS);
+    for (int i = 0; i < LIST_ELEMENTS.length; i++) {
+      // LIST_ELEMENTS[LIST_ELEMENTS.length - 1 - i] iterates LIST_ELEMENTS backwards
+      System.out
+          .println(jedis.lindex(LIST_KEY, i) + " " + LIST_ELEMENTS[LIST_ELEMENTS.length - 1 - i]);
+      assertThat(jedis.lindex(LIST_KEY, i)).isEqualTo(LIST_ELEMENTS[LIST_ELEMENTS.length - 1 - i]);
+    }
+  }
+
+  @Test
+  public void lindex_withNegativeIndexes_returnsElement() {
+    jedis.lpush(LIST_KEY, LIST_ELEMENTS);
+    for (int i = -7; i < 0; i++) {
+      // LIST_ELEMENTS[-(i + 1)] iterates LIST_ELEMENTS forwards
+      assertThat(jedis.lindex(LIST_KEY, i)).isEqualTo(LIST_ELEMENTS[-(i + 1)]);
```

Comment:
For consistency with the above test, and to make it a little easier to parse, could this instead be:
```
    for (int i = 0; i < LIST_ELEMENTS.length; i++) {
      assertThat(jedis.lindex(LIST_KEY, -(i + 1))).isEqualTo(LIST_ELEMENTS[i]);
    }
```